konig-dao-sql-runtime/src/main/java/io/konig/sql/runtime/BigQueryShapeReadService.java

```diff
@@ -115,37 +115,36 @@ private void writeRow(EntityStructure struct, List<FieldValue> row, JsonGenerato
 	private void writeField(JsonGenerator json, FieldInfo fieldInfo, FieldValue field) throws IOException, DaoException {
 		
 		String fieldName = fieldInfo.getName();
-		String fieldType = fieldInfo.getFieldType() == null ? "" : fieldInfo.getFieldType().getLocalName();
-		Object value = field.getValue();
-		
-		if (value instanceof String) {
-			json.writeStringField(fieldName, String.valueOf(formatField(fieldType,value.toString())));
+		String fieldType = fieldInfo.getFieldType() == null ? "" : fieldInfo.getFieldType().toString();
+		Object value = formatField(fieldType,field);	
+
+		if (value instanceof String || value instanceof DateTime) {
+			json.writeStringField(fieldName, value.toString());
+		} else if (value instanceof Integer || value instanceof Long) {
+			json.writeNumberField(fieldName, Integer.parseInt(value.toString()));
 		} else if (value instanceof ArrayList) {
 			List<FieldInfo> fields = fieldInfo.getStruct().getFields();
 			json.writeObjectFieldStart(fieldName);
 			writeArrayField(json, fields, field);
 			json.writeEndObject();
-		} else {
-			throw new DaoException("Field type not supported: " + value.getClass().getName());
+		} else {
+			throw new DaoException("Field type not supported: " + value.getClass().getName());
 		}
 		
 	}
-	private Object formatField(String dataType, String value){
-		dateFormatUTC.setTimeZone(TimeZone.getTimeZone("UTC"));
-		String[] array = {"string","dateTime","int"};		
-		int index = Arrays.asList(array).indexOf(dataType);
-		switch (index) {
-		case 0:				
-			return value.toString();
-		case 1:
-			return
-			 dateFormatUTC.format(new Date(Double.valueOf(value.toString()).longValue() * 1000));
-		case 2:
-			return Integer.parseInt(value.toString());
+	private Object formatField(String fieldType, FieldValue value){
+		switch (fieldType) {
```

Comment:
You also need to include the case "http://www.w3.org/2001/XMLSchema#date" since the MyPedia model uses this data type.